### PR TITLE
Clean up Zs1s2 inputs and outputs and rearrange logic

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
Well, that's a heck of a lot nicer.

File: internal/amos/amos_test.go [hunk 1]
```diff
@@ -147,9 +147,20 @@ func zs1s2test(t *testing.T, x []float64, is []int) {
...
 	}
 
+	zr := complex(input.ZRR, input.ZRI)
+	s1 := complex(input.S1R, input.S1I)
+	s2 := complex(input.S2R, input.S2I)
 	impl := func(input data) data {
-		zrr, zri, s1r, s1i, s2r, s2i, nz, ascle, alim, iuf :=
-			Zs1s2(input.ZRR, input.ZRI, input.S1R, input.S1I, input.S2R, input.S2I, input.NZ, input.ASCLE, input.ALIM, input.IUF)
+		s1, s2, nz, iuf :=
+			Zs1s2(zr, s1, s2, input.ASCLE, input.ALIM, input.IUF)
```
Comment (kortschak) [inline internal/amos/amos_test.go:155]: This would even fit on one line now.

File: internal/amos/amos.go [hunk 2]
```diff
@@ -2169,43 +2177,24 @@ OneNinety:
...
-			// fix once we're sure other changes are correct.
-			c1 -= zr
-			c1 -= zr
-			s1 = cmplx.Exp(c1)
-			as1 = cmplx.Abs(s1)
-			IUF++
-		} else {
-			s1 = 0
-			as1 = 0
+func Zs1s2(zr, s1, s2 complex128, scale, lim float64, iuf int) (s1o, s2o complex128, nz, iufo int) {
```
Comment (kortschak) [inline internal/amos/amos.go:2184]: Why did zro go away?

**Comment by kortschak** [comment]
LGTM

I take it this reduction will be propagated further.

File: internal/amos/amos_test.go (hunk 1 above)
Comment (btracey) [inline internal/amos/amos_test.go:155]: I'll fix in next commit.

File: internal/amos/amos.go (hunk 2 above)
Comment (btracey) [inline internal/amos/amos.go:2184]: Because z isn't modified during the function, so there's no need to return it.

**Comment by btracey** [comment]
Yep, just one chunk at a time.

**Pull request merged**